src/chapters/errors-and-debugging/diagnosing-error-messages.rst

```diff
@@ -0,0 +1,150 @@
+
+
+A Runtime Error
+---------------
+
+Having fixed the syntax error, we can now run our program again. Doing so displays yet another error.
+
+::
+
+   Hello
+   /Users/chris/dev/sandbox/js/syntax.js:1
+   let name = Julie;
+            ^
+
+   ReferenceError: Julie is not defined
+      at Object.<anonymous> (/Users/chris/dev/sandbox/js/syntax.js:1:74)
+      at Module._compile (internal/modules/cjs/loader.js:738:30)
+      at Object.Module._extensions..js (internal/modules/cjs/loader.js:749:10)
+      at Module.load (internal/modules/cjs/loader.js:630:32)
+      at tryModuleLoad (internal/modules/cjs/loader.js:570:12)
+      at Function.Module._load (internal/modules/cjs/loader.js:562:3)
+      at Function.Module.runMain (internal/modules/cjs/loader.js:801:12)
+      at internal/main/run_main_module.js:21:11
+
+
+We have a new error message, this time having to due with line 1 of our code. We didn't see this error before because it is a runtime error, and the program never reached the execution phase earlier. Due to the syntax error on line 2, it stopped during the parsing phase. Even though the current error appears *before* the original syntax error in the code, the syntax error was reported first.
+
+Once again, we are told where the error occurs:
+
+::
+
+   let name = Julie;
+            ^
+
+There appears to be an issue with the assignment statement. You might be able to see what it is, but let's inspect the error message anyway. Doing so will help us understand JavaScript errors more generally.
+
+The message is:
+
+::
+
+   ReferenceError: Julie is not defined
+
+The type of error is ``ReferenceError``. If we search the web for "JS ReferenceError" then one of the first results is the `MDN documentation for ReferenceError <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ReferenceError>`_. The first sentence on this page is:
+
+.. pull-quote:: The ``ReferenceError`` object represents an error when a non-existent variable is referenced.
+
+This information, along with the rest of the message, "Julie is not defined," makes it clear what JavaScript is complaining about.
+
+To us, the issue seems to be that we forgot to enclose the string ``Julie`` in quotes, but this is because we know that we intended to assign the variable ``name`` a string value. However, to JavaScript, there is nothing in the program to indicate that ``Julie`` should be a string. In fact, JavaScript sees ``Julie`` as a variable. Since there is not such defined variable in our program, it complains that there is a ``ReferenceError``. 
```

Comment:
First sentence - "To us, we see that we forgot to enclose the string ``Julie`` in quotes, because we know that we intended to assign the variable ``name`` a string value."

Remove comma after JavaScript in the second sentence.

Last sentence - 'no' rather than 'not', and some shortening.  "Since there is no such defined variable in our program, it returns a ``ReferenceError``. "